`services.TestNodeWatcher` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/4690314237/jobs/8313237564

#### Relevant snippet


```
=== CONT  TestNodeWatcher
{"caller":"services/watcher.go:210","component":null,"level":"debug","message":"Starting watch.","resource-kind":"lock","timestamp":"2023-04-13T14:24:13Z"}
    watcher_test.go:988: 
        	Error Trace:	/__w/teleport/teleport/lib/services/watcher_test.go:988
        	Error:      	Should be empty, but was [Server(name=node0, namespace=default, addr=127.0.0.1:2023, labels=map[])]
        	Test:       	TestNodeWatcher
--- FAIL: TestNodeWatcher (0.02s)
```

--- a/lib/services/watcher_test.go
+++ b/lib/services/watcher_test.go
@@ -1028,7 +1028,7 @@ func TestKubeServerWatcher(t *testing.T) {
 	})
 	require.NoError(t, err)
 	t.Cleanup(w.Close)
-
+	require.NoError(t, w.WaitInitialization())
 	newKubeServer := func(t *testing.T, name, addr, hostID string) types.KubeServer {
 		kube, err := types.NewKubernetesClusterV3(
 			types.Metadata{

--- a/lib/services/watcher_test.go
+++ b/lib/services/watcher_test.go
@@ -958,7 +958,7 @@ func TestNodeWatcher(t *testing.T) {
 	})
 	require.NoError(t, err)
 	t.Cleanup(w.Close)
-
+	require.NoError(t, w.WaitInitialization())
 	// Add some node servers.
 	nodes := make([]types.Server, 0, 5)
 	for i := 0; i < 5; i++ {

